mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/rules/Moves.kt

```diff
@@ -142,3 +147,91 @@ fun Board<Piece<Role>>.diagonals(
     )
   }
 }
+
+/** Moves representing a castling to the left. */
+fun BoardWithHistory<Piece<Role>>.leftCastling() =
+    castling(
+        kingStart = Position(x = 4, y = 7),
```

Comment:
Nitpick: Seems a bit like a code smell https://refactoring.guru/smells/primitive-obsession

Reply:
Not sure

Reply:
Primitive obsession is the fact of using some primitives when a dedicated class may be better suited. I'm not sure how it applies here ? Do you have a more specific suggestion as to what should be changed ?

Reply:
(discussed out-of-band)